tokio/src/runtime/scheduler/current_thread/mod.rs

```diff
@@ -48,6 +49,9 @@ pub(crate) struct Handle {
 
     /// User-supplied hooks to invoke for things
     pub(crate) task_hooks: TaskHooks,
+
+    /// If this is a `LocalRuntime`, flags the owning thread ID.
+    pub(crate) local_tid: Option<ThreadId>,
```

Comment:
Don't we have our own thread id implementation?

Reply:
We do, but our `ThreadId` type is meant more for uniquely identifying runtime threads than for identifying arbitrary threads. Storage and recording of an ID is managed by the `Context`, which is not workable when I am specifically trying to check and see if an arbitrary handle can be used to spawn things locally.